src/nvim/popupmenu.c

```diff
@@ -612,18 +636,33 @@ void pum_redraw(void)
     }
   }
 
+  WinConfig fconfig = WIN_CONFIG_INIT;
+  int pum_border_width = 0 ;
+  if (*p_winborder != NUL) {
+    pum_border_width = 2;
+    fconfig.border = true;
+    Error err = ERROR_INIT;
+    parse_border_style(CSTR_AS_OBJ(p_winborder), &fconfig, &err);
+    if (ERROR_SET(&err)) {
+      emsg(err.msg);
+      api_clear_error(&err);
+      return;
```

Comment:
doing this (parsing and possible fatal error) in the middle of `pum_redraw` seems strange. should we have a `did_set_winborder` in https://github.com/neovim/neovim/blob/97a6259442526e33a73201557b1cb74ccdb64eef/src/nvim/option.c#L2378 which sets some state in popupmenu.c when 'winborder' changes? Then the parsing + error-handling isn't done on every redraw.